src/driver/compile.mli

```diff
@@ -2,20 +2,24 @@ type compiled
 
 val init_stats : Packages.set -> unit
 
-val compile : Fpath.t -> Packages.set -> compiled list
+val compile :
+  ?partial:Fpath.t ->
+  output_dir:Fpath.t ->
+  ?linked_dir:Fpath.t ->
+  Packages.set ->
+  compiled list
+(** Use [partial] to reuse the output of a previous call to [compile]. Useful in
+    the voodoo context.
```

Comment:
It's quite unclear what `partial` means and how to use it. Is the `index.m` file really needed? It seems that it could work without it, at the cost of opening the files in `output_dir`.

Why does the caller have to specify the partial file path ? The compile function could decide where to put it, as it already decides `index.m` which files to read.